tests/integration/test_filter.py

```diff
@@ -122,3 +123,49 @@ def test_filter_year(c):
     expected_df = df[df["year"] < 2016]
 
     assert_frame_equal(expected_df, actual_df)
+
+
+@pytest.mark.parametrize(
+    "query,df_func",
+    [
+        ("SELECT * FROM parquet_ddf WHERE b < 10", lambda x: x[x["b"] < 10]),
+        (
+            "SELECT * FROM parquet_ddf WHERE a < 3 AND (b > 1 AND b < 5)",
+            lambda x: x[(x["a"] < 3) & ((x["b"] > 1) & (x["b"] < 5))],
+        ),
+        (
+            "SELECT * FROM parquet_ddf WHERE (b > 5 AND b < 10) OR a = 1",
+            lambda x: x[((x["b"] > 5) & (x["b"] < 10)) | (x["a"] == 1)],
+        ),
+        (
+            "SELECT a FROM parquet_ddf WHERE (b > 5 AND b < 10) OR a = 1",
+            lambda x: x[((x["b"] > 5) & (x["b"] < 10)) | (x["a"] == 1)][["a"]],
+        ),
+        pytest.param(
+            "SELECT * FROM parquet_ddf WHERE year(d) < 2015",
+            lambda x: x[x["d"].dt.year < 2015],
+            # This test will fail, because the filters will not get pushed
+            # down to read_parquet. However, the query should still succeed.
+            marks=pytest.mark.xfail(
+                reason="Predicate pushdown does not support datetime accessors."
+            ),
+        ),
+    ],
+)
+def test_predicate_pushdown(c, parquet_ddf, query, df_func):
+
+    # Check for predicate pushdown.
+    # We can use the `hlg_layer` utility to make sure the
+    # `filters` field has been populated in `creation_info`
+    return_df = c.sql(query)
+    assert hlg_layer(return_df.dask, "read-parquet").creation_info["kwargs"]["filters"]
+
+    # Check computed result is correct.
+    # Note that we must sort by index to ensure
+    # ordering can be compared directly
+    return_df = return_df.compute()
+    df = parquet_ddf.compute()
+    expected_df = df_func(df)
+    assert_frame_equal(
+        return_df.sort_index(), expected_df.sort_index(),
+    )
```

Comment:
Would it be possible to do this assertion using `assert_eq`? Or do we need a serial dataframe for `df_func`?

```suggestion
    expected_df = df_func(parquet_ddf)
    assert_frame_equal(
        return_df.sort_index(), expected_df.sort_index(),
    )
```

For context, trying to move over to `assert_eq` wherever possible in https://github.com/dask-contrib/dask-sql/pull/365 🙂

Reply:
Yep - `assert_eq` makes sense, but I was just going with the pattern used by other tests :)